yt_dlp/extractor/rplaylive.py

```diff
@@ -0,0 +1,321 @@
+        video_id = self._match_id(url)
+
+        playlist_id = traverse_obj(parse_qs(url), ('playlist', ..., any))
+        if playlist_id and self._yes_playlist(playlist_id, video_id):
+            playlist_info = self._download_json(
+                'https://api.rplay.live/content/playlist', playlist_id,
+                query={'playlistOid': playlist_id, **self.requestor_query},
+                headers=self.jwt_header, fatal=False)
+            if playlist_info:
+                entries = traverse_obj(playlist_info, ('contentData', ..., '_id', {
+                    lambda x: self.url_result(f'https://rplay.live/play/{x}/', ie=RPlayVideoIE, video_id=x)}))
+                return self.playlist_result(entries, playlist_id, playlist_info.get('name'))
+            else:
+                self.report_warning('Failed to get playlist, downloading video only')
+
+        video_info = self._download_json('https://api.rplay.live/content', video_id, query={
+            'contentOid': video_id,
+            'status': 'published',
+            'withComments': True,
+            'requestCanView': True,
+            **self.requestor_query,
+        }, headers=self.jwt_header)
+        if video_info.get('drm'):
+            raise ExtractorError('This video is DRM-protected')
+
+        metainfo = traverse_obj(video_info, {
+            'title': ('title', {str}),
+            'description': ('introText', {str}),
+            'release_timestamp': ('publishedAt', {parse_iso8601}),
+            'timestamp': ('createdAt', {parse_iso8601}),
+            'duration': ('length', {float_or_none}),
+            'uploader': ('creatorInfo', 'nickname', {str}),
+            'uploader_id': ('creatorOid', {str}),
+            'tags': ('hashtags', lambda _, v: v[0] != '_'),
+            'age_limit': (('hideContent', 'isAdultContent'), {lambda x: 18 if x else None}, any),
+            'live_status': ('isReplayContent', {lambda x: 'was_live' if x else None}),
+        })
+
+        m3u8_url = traverse_obj(video_info, ('canView', 'url', {url_or_none}))
+        if not m3u8_url:
+            msg = 'You do not have access to this video'
+            if traverse_obj(video_info, ('viewableTiers', 'free')):
+                msg = 'This video requires a free subscription to access'
+            if not self.user_id:
+                # credential is in browser localStorage only, no cookies to use
+                msg += f'. {self._login_hint(method="password")}'
+            raise ExtractorError(msg, expected=True)
+
+        formats = self._extract_m3u8_formats(m3u8_url, video_id, headers={
+            'Referer': 'https://rplay.live/', 'Butter': self.get_butter_token()})
+        for fmt in formats:
+            m3u8_doc = self._download_webpage(fmt['url'], video_id, 'getting m3u8 contents', headers={
+                'Referer': 'https://rplay.live/', 'Butter': self.get_butter_token()})
+            fmt['url'] = encode_data_uri(m3u8_doc.encode(), 'application/x-mpegurl')
```

Comment:
This can produce data URIs that exceed `MAX_ARG_STRLEN` on Linux and thus cannot be passed to ffmpeg. Can the `url` only be fetched once? If not, you can get the HLS key here and return it along with `Butter`, so ffmpeg handles it all.

Otherwise the m3u8 file needs to be written to a temporary location and then passed to ffmpeg like that.

Reply:
ffmpeg will not work as an external downloader for this site; there's no way to pass the `hls_aes` key to ffmpeg

Reply:
Okay, after some messing around it seems this extractor makes `pycryptodomex` a dependency of `yt-dlp`.

Reply:
pycryptodome/x is an optional dependency of yt-dlp that is required by some extractors, yes